src/main/java/phi/Parser.java

```diff
@@ -0,0 +1,115 @@
+package phi;
+
+/**
+ * Represents the handling of user-commands and making sense of them
+ */
+public class Parser {
+    private final TaskList tasks;
+
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Method that handles parsing user inputs into its respective commands. Helps to check for user-input errors.
+     * String output will be handled by the function caller (PHI)
+     *
+     * @param input User-input given to PHI
+     * @return      Response of the parser to the input, to be printed by PHI
+     */
+    public String handle(String input) {
```

Comment:
Consider giving the function a more descriptive name, such as handleInput() or parseInput() to improve readability. 😄